01_fib/CameronSchafer/fib_test.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+//main() test
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote(`1
+1
+2
+3
+5
+8
+13
+`)
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()\nexpected: %q\nactual: %q", expected, actual)
+	}
+}
+
+func TestFibOutput(t *testing.T) {
+	//tempBuf is used to reset the output buffer for each test
+	var buf, tempBuf bytes.Buffer
+	out = &buf
+
+	//test cases with descriptions.
+	testCases := []struct {
+		description string
+		input       int
+		expected    string
+	}{
+		{description: "fib 7", input: 7,
+			expected: strconv.Quote(`1
+1
+2
+3
+5
+8
+13
+`),
+		},
+		{description: "fib 1", input: 1,
+			expected: strconv.Quote("1\n"),
+		},
+		{description: "fib -7", input: -7,
+			expected: strconv.Quote(`1
+-1
+2
+-3
+5
+-8
+13
+`),
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
```

Comment:
Why do we need to do this?

Reply:
Without this assignment, test was not accessible within the for loop.

Reply:
Is it something to do with the anonymous function/closure?

Reply:
To be honest, I'm not exactly sure. I think it has something to do with the way I was trying to access the struct array. I will look into it some more and get back to you.

Reply:
It is a linting error that pops up if I don't do that assignment:
`Using the variable on range scope 'test' in function literal (scopelint)`

So I feel it is safe to just do the assignment to remove the error.

Reply:
@CameronSchafer I think I found the [answer](https://stackoverflow.com/questions/26692844/captured-closure-for-loop-variable-in-go)

When your function literal captures the `test` variable it is capturing a `reference to the value` not the `value itself`.  So if the functions are executed outside of the for loop then they will have the last value of the range not the value at the time of iteration.

So you have done the right thing here 👍

Reply:
@nicholascross Oh Ok. That makes sense now. Thank you